agent/command/s3_util.go

```diff
@@ -78,3 +79,15 @@ func getS3OpBackoff() *backoff.Backoff {
 func shouldRunForVariant(bvs []string, bv string) bool {
 	return len(bvs) == 0 || utility.StringSliceContains(bvs, bv)
 }
+
+// getAssumedRoleARN checks if the provided session token
+// is associated with an assumed role. If it is, it returns
+// the role ARN.
+func getAssumedRoleARN(conf *internal.TaskConfig, sessionToken string) string {
+	if sessionToken != "" {
```

Comment:
nit: This is really tiny, but in Go, it's generally preferable to return early over nesting ifs and the `ok` isn't needed, so this could be rewritten more simply as:
```
if sessionToken == "" {
    return ""
}

return conf.AssumeRoleRoles[sessionToken]
```